api/src/main/java/io/grpc/InternalMetadata.java

```diff
@@ -82,4 +82,19 @@ public static Metadata newMetadata(int usedNames, byte[]... binaryValues) {
   public static int headerCount(Metadata md) {
     return md.headerCount();
   }
+
+  @Internal
+  public static Object[] serializePartial(Metadata md) {
+    return md.serializePartial();
+  }
+
+  @Internal
+  public static <T> Object parsedValue(Key<T> key, T value) {
+    return Metadata.LazyValue.create(key, value);
+  }
+
+  @Internal
+  public static Metadata newMetadataWithParsedValues(int usedNames, Object[] namesAndValues) {
```

Comment:
As these methods are going to be used in other packages than `grpc-api`, better to have a javadoc, e.g. document the restriction on the argument `Object[] namesAndValues`.

Reply:
Done.